### DISCUSSION TITLE
是否可以考虑让 validator function 内能够访问到 field？

### DISCUSSION THREAD

**Discussion by JustDs** [alibaba/formily#1584] [Ideas]
业务场景下会有一些较复杂的联动校验，但是 validator function 只知道值是什么，而不知道当前字段是哪个字段。

是否可以考虑吧 field 放到 context 中，让 validator function 能够获知这一信息？
https://github.com/alibaba/formily/blob/d14c042ec1b6c81abf2b9e3603e760b6bcdd7c31/packages/core/src/shared/internals.ts#L193-L197

不确定这么做会不会有一些 bad case，希望讨论下。

**Comment by janryWang** [comment]
不会支持，因为validator 是纯函数，不推荐依赖外部副作用，如果有依赖，会导致校验时机很不可控，所以存在副作用的推荐在effects或reactions里处理

**Reply by JustDs** [reply]
通过 reactions 去手动 set errors 吗？

**Reply by janryWang** [reply]
对
